src/implementation/Client/GRPCClient/crypto.ts

```diff
@@ -0,0 +1,178 @@
+
+  constructor(client: GRPCClient) {
+    this.client = client;
+  }
+
+  encrypt(opts: EncryptRequest): Promise<Duplex>;
+  encrypt(inData: Buffer | ArrayBuffer | ArrayBufferView | string, opts: EncryptRequest): Promise<Buffer>;
+  async encrypt(
+    arg0: Buffer | ArrayBuffer | ArrayBufferView | string | EncryptRequest,
+    opts?: EncryptRequest,
+  ): Promise<Duplex | Buffer> {
+    // Handle overloading
+    // If we have a single argument, assume the user wants to use the Duplex stream-based approach
+    let inData: Buffer | undefined;
+    if (opts === undefined) {
+      opts = arg0 as EncryptRequest;
+    } else {
+      // Throws if arg0 is not a supported type
+      inData = this.toArrayBuffer(arg0);
+    }
+
+    // Ensure required options are present
+    if (!opts) {
+      throw new Error(`Parameter 'opts' must be defined`);
+    }
+    if (!opts.componentName) {
+      throw new Error(`Option 'componentName' is required`);
+    }
+    if (!opts.keyName) {
+      throw new Error(`Option 'keyName' is required`);
+    }
+    if (!opts.keyWrapAlgorithm) {
+      throw new Error(`Option 'keyWrapAlgorithm' is required`);
+    }
+
+    // Create the gRPC stream
+    const client = await this.client.getClient();
+    const grpcStream = client.encryptAlpha1();
+
+    // Create a duplex stream that will send data to the server and read from it
+    const duplexStream = new DaprChunkedStream(grpcStream, pbEncryptRequest, (req) => {
+      // Disable the @typescript-eslint/no-non-null-assertion linter in this block because opts here is guaranteed to be non-nil (see the check above), but TS doesn't seem to believe that
+      /* eslint-disable @typescript-eslint/no-non-null-assertion */
+      const reqOptions = new pbEncryptRequestOptions();
+      reqOptions.setComponentName(opts!.componentName);
+      reqOptions.setKeyName(opts!.keyName);
+      reqOptions.setKeyWrapAlgorithm(opts!.keyWrapAlgorithm);
+      if (opts!.dataEncryptionCipher) {
+        reqOptions.setDataEncryptionCipher(opts!.dataEncryptionCipher);
+      }
+      if (opts!.decryptionKeyName) {
+        reqOptions.setDecryptionKeyName(opts!.decryptionKeyName);
+      }
+      if (opts!.omitDecryptionKeyName) {
+        reqOptions.setOmitDecryptionKeyName(opts!.omitDecryptionKeyName);
+      }
+      req.setOptions(reqOptions);
+      /* eslint-enable @typescript-eslint/no-non-null-assertion */
+    });
+
+    // Process the data
+    return this.processStream(duplexStream, inData);
+  }
+
+  decrypt(opts: DecryptRequest): Promise<Duplex>;
+  decrypt(inData: Buffer | ArrayBuffer | ArrayBufferView, opts: DecryptRequest): Promise<Buffer>;
+  async decrypt(
+    arg0: Buffer | ArrayBuffer | ArrayBufferView | DecryptRequest,
+    opts?: DecryptRequest,
+  ): Promise<Duplex | Buffer> {
+    // Handle overloading
+    // If we have a single argument, assume the user wants to use the Duplex stream-based approach
+    let inData: Buffer | undefined;
+    if (opts === undefined) {
```

Comment:
Shouldn't we use `!opts`?

Reply:
I like explicitly checking for "undefined" in these cases as it's not ambiguous.